pyvista/plotting/colors.py

```diff
@@ -348,10 +377,473 @@
+        if h.startswith('0x'):
+            h = h[2:]
+        return h
+
+    @staticmethod
+    def convert_color_channel(val: Union[int, np.integer, float, np.floating, str]) -> int:
+        """Convert the given color channel value to the integer representation (values between ``0`` and ``255``).
+
+        Parameters
+        ----------
+        val : int, float or str
+            Color channel value to convert. Supported input values are a hex string of length 2 (``'00'`` to ``'ff'``),
+            a float (``0.0`` to ``1.0``) or an integer (``0`` to ``255``).
+
+        Returns
+        -------
+        int
+            Color channel value in the integer representation.
+
+        """
+        if isinstance(val, str):
+            # From hexadecimal value
+            val = int(Color.strip_hex_prefix(val), 16)
+        elif np.issubdtype(np.asarray(val).dtype, np.floating) and np.ndim(val) == 0:
+            # From float
+            val = int(round(255 * val))
+        if (
+            np.issubdtype(np.asarray(val).dtype, np.integer)
+            and np.ndim(val) == 0
+            and 0 <= val <= 255
+        ):
+            # From integer
+            return int(val)
+        else:
+            raise ValueError(f"Unsupported color channel value provided: {val}")
+
+    def _from_rgba(self, rgba):
+        """Construct color from an RGB(A) sequence."""
+        if len(rgba) == 3:
+            # Keep using current opacity if it is not provided.
+            rgba = [*rgba, self._opacity]
+        try:
+            if len(rgba) != 4:
+                raise ValueError("Invalid length for RGBA sequence.")
+            self._red, self._green, self._blue, self._opacity = [
+                self.convert_color_channel(c) for c in rgba
+            ]
+            self._name = None
+        except ValueError as e:
+            raise ValueError(f"Invalid RGB(A) sequence: {rgba}") from e
+
+    def _from_hex(self, h):
+        """Construct color from a hex string."""
+        h = self.strip_hex_prefix(h)
+        try:
+            self._from_rgba([self.convert_color_channel(h[i : i + 2]) for i in range(0, len(h), 2)])
+        except ValueError as e:
+            raise ValueError(f"Invalid hex string: {h}") from e
+
+    def _from_str(self, n: str):
+        """Construct color from a name or hex string."""
+        n = n.lower()
+        if len(n) == 1:
+            # Single character
+            # Convert from single character to full hex
+            if n not in color_char_to_word:
+                raise ValueError(
+                    'Single character string must be one of the following:'
+                    f'\n{str(color_char_to_word.keys())}'
+                )
+            n = color_char_to_word[n]
+            self._from_hex(hexcolors[n])
+            self._name = n
+        elif n in hexcolors:
+            # Color name
+            self._from_hex(hexcolors[n])
+            self._name = n
+        elif n in 'paraview' or n in 'pv':
```

Comment:
This was all taken directly from here

https://github.com/pyvista/pyvista/blob/2df0690cea486a60dd747b8d7c477f15e61e2425/pyvista/plotting/colors.py#L388

Not sure what the original intent was there.

Reply:
I'm pretty sure that was also a bug. This would mean that the string `'p'` would match, or the string `'par'` if we exclude single-character codes earlier. I think this was meant to be an either-or check.

Reply:
Resolved using the color synonyms.